### ISSUE TITLE
Performance regression in 4.1 because of aa=1 correctness & Akamai

### ISSUE THREAD

**Issue by ahupowerdns** [PowerDNS/pdns#5874]
- Program: Recursor

### Short description
In the awful CNAME chain of www.kia.com, we end up at:
```
$ dig www.kia.com.edgekey.net.globalredir.akadns.net +norecurs @a12-131.akadns.org

; <<>> DiG 9.10.3-P4-Ubuntu <<>> www.kia.com.edgekey.net.globalredir.akadns.net +norecurs @a12-131.akadns.org
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 56864
;; flags: qr; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
;; QUESTION SECTION:
;www.kia.com.edgekey.net.globalredir.akadns.net.	IN A

;; ANSWER SECTION:
www.kia.com.edgekey.net.globalredir.akadns.net.	3600 IN	CNAME e2884.x.akamaiedge.net.

;; Query time: 7 msec
;; SERVER: 184.26.160.131#53(184.26.160.131)
;; WHEN: Sun Oct 29 20:23:02 CET 2017
;; MSG SIZE  rcvd: 108
```
Note lack of aa bit. However, because this is Akamai, we have to still work with this. What happens next is that we cache the CNAME delegation with 'auth=0'. 

For the next query that passes by this CNAME, we demand an auth=1 cache entry. Which is not available, so we ignore the cache and query Akamai again.

This leads to unhappy users of 'www.kia.com' because every 20 seconds they pay a latency penalty.

>

### Steps to reproduce
Start recursor from git master. Query www.kia.com, wait 20 seconds, query again, observe stupidly slow response. Or, disable the packet cache and observe a stupidly slow response _every time_.

### Expected behaviour
I would expect Akamai not to serve such broken responses. However, since they do, and since they are so huge, we should ponder working around this.

### Actual behaviour
We actually get a response, but slower than we should.

### Other information
This is new in 4.1.

**Comment by ahupowerdns** [comment]
In terms of solutions, I'm tempted to put this behind something with a bad name 'accept-unauth-cname', and when we do accept such an unauth CNAME, promote it to aa=1 so as not to hurt downstream processing.  Another name for the feature that has been proposed is 'akamai-me-harder'.

**Comment by ahupowerdns** [comment]
Note: We've contacted Akamai, and we hope they'll fix this before we release 4.1.

**Comment by ahupowerdns** [comment]
This will likely be fixed within the Akamai CDN.
